gemini_api.py

```diff
@@ -0,0 +1,126 @@
+"""Gemini API functions."""
+from typing import List, Tuple, Callable
+import os
+from google import genai
+from google.genai import types
+from dotenv import load_dotenv
+import utils
+
+
+load_dotenv()
+
+client = genai.Client(api_key=os.getenv("GOOGLE_API_KEY"))
+
+
+grounding_tool = types.Tool(
+    google_search=types.GoogleSearch()
+)
+
```

Comment:
In general, it is not a good idea to put commands directly into the file - they get executed everytime they are imported. In your code, this happens exactly once, so it runs as expected, but often as projects grow you might need some of the functions defined here elsewhere.